pandas/tests/indexes/datetimes/test_timezones.py

```diff
@@ -1012,6 +1012,22 @@ def test_iteration_preserves_nanoseconds(self, tz):
         for i, ts in enumerate(index):
             assert ts == index[i]
 
+    @pytest.mark.parametrize('arg, expected_arg', [
+        [[], []],
+        [date_range('2018-11-04', periods=4, freq='H', tz='US/Pacific'),
```

Comment:
can you try with a dateutil as well (I think we default to pytz)

Reply:
@pganssle Do dateutil timezones provide a way to access the possible daylight savings time shifts like `pytz.timezone._transition_info` does?

```
# I'm interested in the 2nd element of the tuples below
In [4]: tz._transition_info
Out[4]:
[(datetime.timedelta(-1, 58020), datetime.timedelta(0), 'LMT'),
 (datetime.timedelta(-1, 57600), datetime.timedelta(0), 'PST'),
 (datetime.timedelta(-1, 61200), datetime.timedelta(0, 3600), 'PDT'),
...
```
Context: I am trying to create a `DatetimeIndex.is_dst` method that indicates whether a particular timestamp is in daylight savings time. I know it's possible to do `bool(datetime.dst())` on each timestamp of the `DatetimeIndex`, but I am curious if there's a more efficient way determine DST for a dateutil timezone than iterating.

Reply:
@mroeschke Yes and no. Yes in the sense that this is Python and obviously `dateutil` has to store that information *somewhere*. No in the sense that neither `dateutil` *nor* `pytz` exposes that information publicly. Note that `_transition_info` is a private member.

That said:

1. I'm not entirely sure why you want to implement a "fast isdst" - I'm not sure I've ever heard of a case where it was useful to know whether or not a time had or did not have DST.
2. I don't really see what this would buy you - the member you list looks like what you would get from `utcoffset`, `dst` and `tzname`, respectively. How do you expect to make use of this information?